solidoc.json

```diff
@@ -3,36 +3,97 @@
   "outputPath": "./contract_docs",
   "noCompilation": true,
   "compiler": "truffle compile",
-  "ignoreFiles": [ "AccessControlUpgradeable.json",
+  "ignoreFiles": [ "AccessControl.json",
+    "AccessController.json",
+    "AccessControlUpgradeable.json",
     "AddressUpgradeable.json",
+    "AgentHandlerFacet.json",
+    "BeaconClientBase.json",
+    "BeaconClientLib.json",
+    "BeaconClientProxy.json",
+    "BeaconClientBeacon.json",
+    "BundleBase.json",
+    "BundleHandlerFacet.json",
+    "BuyerHandlerFacet.json",
+    "BosonToken.json",
+    "BosonVoucher.json",
+    "ClientBase.json",
+    "ClientLib.json",
+    "ClientProxy.json",
+    "ClientExternalAddressesBase",
+    "ConfigHandlerFacet.json",
+    "Context.json",
     "ContextUpgradeable.json",
     "BosonToken.json",
+    "EIP712.json",
+    "EIP712Lib.json",
+    "ERC165Facet.json",
     "ERC165Upgradeable.json",
     "ERC20.json",
+    "ERC721.json",
     "ERC20Pausable.json",
+    "ERC721EnumerableUpgradeable.json",
     "ERC721Upgradeable.json",
     "ERC1155Upgradeable.json",
     "FallbackError.json",
     "Foreign20.json",
+    "Foreign20Malicious.json",
+    "Foreign20Malicious2.json",
+    "Foreign20TransferFromReturnFalse.json",
+    "Foreign20TransferReturnFalse.json",
     "Foreign20NoName.json",
+    "Foreign20WithFee.json",
     "Foreign721.json",
     "Foreign1155.json", 
     "IERC20.json",
     "IERC20Metadata.json",
     "IERC165.json",
+    "IERC721.json",
+    "IERC1155.json",
+    "IERC2981Upgradeable.json",
     "IERC165Upgradeable.json",
     "IERC721MetadataUpgradeable.json",
     "IERC721ReceiverUpgradeable.json",
+    "IERC721EnumerableUpgradeable.json",
     "IERC721Upgradeable.json",
     "IERC1155MetadataURIUpgradeable.json",
     "IERC1155ReceiverUpgradeable.json",
     "IERC1155Upgradeable.json",
     "Initializable.json",
+    "JewelerLib.json",
+    "Math.json",
+    "MetaTransactionsHandlerFacet.json",
+    "MockEIP712Base.json",
+    "MockExchangeHandlerFacet.json",
+    "MockExchangeHandlerFacetWithDefect.json",
+    "MockMetaTransactionsHandlerFacet.json",
+    "MockNativeMetaTransaction.json",
+    "MockNFTAuth721.json",
+    "OfferBase.json",
+    "OfferHandlerFacet.json",
+    "Ownable.json",
+    "OrchestrationHandlerFacet.json",
+    "OwnableUpgradeable.json",
     "Pausable.json",
+    "PausableBase.json",
+    "PauseHandlerFacet.json",
+    "ProtocolBase.json",
+    "ProtocolDiamond.json",
+    "ProtocolLib.json",
+    "Proxy.json",
+    "ReentrancyGuardBase.json",
+    "SafeERC20.json",
+    "SellerBase.json",
+    "SellerHandlerFacet.json",
+    "Strings.json",
     "StringsUpgradeable.json",
     "Test1Facet.json",
     "Test2Facet.json",
+    "Test2FacetUpgrade.json",
     "Test3Facet.json",
+    "TestFacet256.json",
     "TestFacetLib.json",
-    "Test2FacetUpgrade.json" ]
+    "TestInitializableDiamond.json",
+    "Test2FacetUpgrade.json",
+    "TestProtocolFunctions.json" ]
 }
```

Comment:
There is a new folder `contracts/example/` which has the snapshot gate example and support contracts. You may need to omit these as well.

Reply:
Also, the ignorelist seems like it could get out of control. Is there no better way of specifying what to include?

Reply:
`solidoc` currently works on an exclusion basis... let me take a look to see if I can reverse it so that it works off an include-list

Reply:
MIght be nice to have either mode available. Depending on whether the list to include would be longer that the one to exclude or visa versa.

Reply:
I've found a more sensible way to do this - we can specify the directories to compile in the `package.json` and that also allows us to then use the `solidoc.json` to exclude files within this. So currently the change to do this is in this repo rather than touching `solidoc`.